odoo/osv/expression.py

```diff
@@ -1000,7 +1011,18 @@ def push_result(sql):
             # -> else: crash
             # ----------------------------------------
 
-            if field.inherited:
+            if  not field.relational and \
+                field.store and \
+                operator in WILDCARD_OPERATORS and \
+                pycompat.to_text(right) == '':
+
+                # The righ hand of the leaf is an empty string and the operator is
+                # a wildcard operator. The resulting wildcard will always be "%%"
+                # which is equivalent to TRUE or FALSE depending on the operator.
+                sql_field = model._field_to_sql(alias, left, self.query)
+                push_result(SQL("(%s IS NULL)", sql_field) if operator in NEGATIVE_TERM_OPERATORS else SQL("TRUE"))
```

Comment:
Is it to avoid repeating the logic twice (for translate and normal field)? Is it cleaner to move the logic there (even if it duplicates the code a bit)?

Also it means that the condition `(need_wildcard and not right)` become useless ?